compiler/quilt/test/test_import.py

```diff
@@ -198,6 +198,130 @@ def test_multiple_package_dirs(self):
+
+        # First and second package in the new build dir.
+        with patch.dict(os.environ, {'QUILT_PRIMARY_PACKAGE_DIR': new_build_dir}):
+            command.build('test:foo/multiple1', simple_build_path)
+            command.build('test:foo/multiple2', simple_build_path)
+
+        # Cannot see the second package yet.
+        with self.assertRaises(ImportError):
+            from quilt.team.test.foo import multiple2
+
+        # Now search the new build dir.
+        dirs = 'foo/%s:%s' % (PACKAGE_DIR_NAME, new_build_dir)
+        with patch.dict(os.environ, {'QUILT_PACKAGE_DIRS': dirs}):
+            # Can import the second package now.
+            from quilt.team.test.foo import multiple2
+
+            # The first package contains data from the default dir.
+            from quilt.team.test.foo import multiple1
+            assert multiple1.dataframes
+
+            # Check search order again just to make sure it doesn't return same named empty package
+            from quilt.team.test.foo import multiple1
+            assert multiple1.dataframes
+
+            teststore = PackageStore(self._store_dir)
+            contents = open(os.path.join(teststore.package_path('test', 'foo', 'multiple1'),
+                                          Package.CONTENTS_DIR,
+                                          multiple1._package.get_hash())).read()
+            assert contents
+
+            # bad imports
+            with self.assertRaises(ImportError):
+                from quilt.team.test.nonexisting import multiple2
+            with self.assertRaises(ImportError):
+                from quilt.team.test.foo import nonexistingdata
+            with self.assertRaises(ImportError):
+                import quilt.team.test.foo.nonexisting
+            with self.assertRaises(ImportError):
+                import quilt.team.test.bad_user.nonexisting
+            with self.assertRaises(ImportError):
+                from quilt.team.test.foo.dataframes import blah
+            with self.assertRaises(ImportError):
+                from quilt.team.test.foo.baz import blah
+
+        # add same dir to the search path
+        dirs = 'foo/%s:%s:%s' % (PACKAGE_DIR_NAME, new_build_dir, new_build_dir)
+        with patch.dict(os.environ, {'QUILT_PACKAGE_DIRS': dirs}):
+            from quilt.team.test.foo import multiple2
+
+        # check that invalid folder names fail
+        bad_build_dir = '?\*;:|<>'
+        with patch.dict(os.environ, {'QUILT_PRIMARY_PACKAGE_DIR': bad_build_dir}):
+            with self.assertRaises(AssertionError):
+                command.build('test:bar/multiple1', simple_build_path)
+        bad_build_dir = ''
+        with patch.dict(os.environ, {'QUILT_PRIMARY_PACKAGE_DIR': bad_build_dir}):
+            with self.assertRaises(AssertionError):
+                command.build('test:bar/multiple1', simple_build_path)
+
+    def test_multiple_package_dirs_with_subpackage(self):
```

Comment:
Not really sure what subpackage means here - but I think this tests pretty much the same thing as `test_multiple_package_dirs`